src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -0,0 +1,611 @@
+                                        }
+                                    }
+
+                                    //add code snippets/changes.
+                                    if (file.Methods != null && file.Methods.Any())
+                                    {
+                                        //code changes are chunked together by methods. Easier for document node modifications.
+                                        foreach (var method in file.Methods)
+                                        {
+                                            var methodName = method.Key;
+                                            var methodChanges = method.Value;
+
+                                            if (!string.IsNullOrEmpty(methodName) &&
+                                                methodChanges != null &&
+                                                methodChanges.CodeChanges != null)
+                                            {
+                                                //get method
+                                                IDictionary<string, string>? parameterValues = null;
+                                                var nodes = modifiedClassDeclarationSyntax?.DescendantNodes();
+                                                var methodNode = modifiedClassDeclarationSyntax?.DescendantNodes().Where(
+                                                        node => node is MethodDeclarationSyntax mds &&
+                                                        mds.Identifier.ValueText.Contains(methodName) &&
+                                                        (parameterValues = VerfiyParameters(methodChanges.Parameters, mds.ParameterList.Parameters.ToList())) != null).FirstOrDefault();
+
+                                                if (methodNode != null)
+                                                {
+                                                    //get method block
+                                                    var blockSyntaxNode = methodNode.DescendantNodes().OfType<BlockSyntax>().FirstOrDefault();
+                                                    var modifiedBlockSyntaxNode = methodNode.DescendantNodes().OfType<BlockSyntax>().FirstOrDefault();
+
+                                                    foreach (var change in methodChanges.CodeChanges)
+                                                    {
+                                                        if (!string.IsNullOrEmpty(change.Block))
+                                                        {
+                                                            //using defaults for leading and trailing trivia
+                                                            var trailingTrivia = new SyntaxTriviaList(SyntaxFactory.CarriageReturnLineFeed);
+                                                            var leadingTrivia = new SyntaxTriviaList(SyntaxFactory.Whitespace("    "));
```

Comment:
>"    " [](http://example.com/codeflow?start=126&length=6)

hardcoded spaces?

Reply:
currently going with Whitespaces. Will have an item to improve based on user's preferences of tabs or whitespaces.